From b209bf23bc97b7758b9dc8f68d2a835295960226 Mon Sep 17 00:00:00 2001
From: Brian Belleville <bbellevi@uci.edu>
Date: Tue, 20 Feb 2018 14:54:25 -0800
Subject: [PATCH] floppy: Don't print kernel addresses to log in show_floppy

Outputting kernel addresses will reveal the locations of kernel code
and data. Change the cases in show_floppy that print
fd_timer.work.func and fd_timeout.work.func to use the %pf format
specifier, which will print the symbol name, like what is done for the
other function pointers printed by show_floppy. No longer output the
value of cont. The variable cont is a pointer that can hold the
address of kernel global variables.

Signed-off-by: Brian Belleville <bbellevi@uci.edu>
---
 drivers/block/floppy.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/block/floppy.c b/drivers/block/floppy.c
index eae484acfbbc..e29d4174ea2c 100644
--- a/drivers/block/floppy.c
+++ b/drivers/block/floppy.c
@@ -1819,15 +1819,14 @@ static void show_floppy(void)
 	if (work_pending(&floppy_work))
 		pr_info("floppy_work.func=%pf\n", floppy_work.func);
 	if (delayed_work_pending(&fd_timer))
-		pr_info("delayed work.function=%p expires=%ld\n",
+		pr_info("delayed work.function=%pf expires=%ld\n",
 		       fd_timer.work.func,
 		       fd_timer.timer.expires - jiffies);
 	if (delayed_work_pending(&fd_timeout))
-		pr_info("timer_function=%p expires=%ld\n",
+		pr_info("timer_function=%pf expires=%ld\n",
 		       fd_timeout.work.func,
 		       fd_timeout.timer.expires - jiffies);
 
-	pr_info("cont=%p\n", cont);
 	pr_info("current_req=%p\n", current_req);
 	pr_info("command_status=%d\n", command_status);
 	pr_info("\n");
-- 
2.16.2

